memcached.c

```diff
@@ -5793,11 +5800,11 @@ static void process_flush_all_command(conn *c, token_t *tokens, const size_t nto
     }
 
     if (ntokens != (c->noreply ? 3 : 2)) {
-        exptime = strtol(tokens[1].value, NULL, 10);
-        if(errno == ERANGE) {
-            out_string(c, "CLIENT_ERROR bad command line format");
+        if (!safe_strtol(tokens[1].value, &exptime_int)) {
+            out_string(c, "CLIENT_ERROR invalid exptime argument");
             return;
         }
+        exptime = realtime(EXPTIME_TO_POSITIVE_TIME(exptime_int));
```

Comment:
Note the comment below: realtime() is run after exptime is checked for a 0 value.

Reply:
OK. I fixed it.